src/index.ts

```diff
@@ -730,10 +731,8 @@ client = Client()`;
     cluster: IClusterModel,
     editor: CodeEditor.IEditor
   ): void {
-    const cursor = editor.getCursorPosition();
-    const offset = editor.getOffsetAt(cursor);
     const code = getClientCode(cluster);
-    editor.model.value.insert(offset, code);
+    editor.model.sharedModel.setSource(code);
```

Comment:
I need to familiarize myself a bit more with ydoc, but it seems like [`updateSource`](https://jupyter-ydoc.readthedocs.io/en/latest/api/classes/YBaseCell.html#updateSource) would be less destructive if there is  existing text in a cell.

Reply:
Makes sense. I will test it before making the change!

Reply:
We are always [inserting a new cell](https://github.com/dask/dask-labextension/blob/73c45ce5f726dbdac5bcfa53ed99cee5f8ab1256/src/index.ts#L792) to inject this code. So, `updateSource` would not make a lot of difference.

Its good that you pointed it out. Seems like we need to wait for the cell to be ready before editor is available. Probably this is JLab 4 change too. I have fixed it but if you think we can do it better, please let me know.

Reply:
> We are always [inserting a new cell](https://github.com/dask/dask-labextension/blob/73c45ce5f726dbdac5bcfa53ed99cee5f8ab1256/src/index.ts#L792) to inject this code. So, updateSource would not make a lot of difference.

Interesting, I had forgotten that, but right you are! https://github.com/dask/dask-labextension/commit/a77b0b45ec3457586e2f55ab2030ad81e556ffb9

In the case of a console or code editor we are not doing that, but I don't know if too many people are doing that in practice, so I'm not too worried about it in practice. If anybody complains, it's easy enough to fix.